### PR TITLE
fix regex matching for prometheus retention timespan

### PR DESCRIPTION
**Pull request by weirdwiz**
[Playground Link](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBNBOBGGBeGADAhgGwKbygFtMxMBzfAWUwEswAuAKBhgDcRsBXQ3AYW1qEAKrkIAHAVFxMWLHlEwATTApmyWpHvRiEAngFoc+IiXL59wATULNZEMbmBr18XBBCd4wN8-UxXAI6cblAQvn5wUCDwZtoALAAMAOI0turQ0Wb8mBAQAHKYWjAgwBD6GTEUlpzQ5t5iABb68ABGioxi8CDyDbg1ANIAHGG2rlJgUDTg2gBMcYozDbbsXDzZ1iLiktJpOrgKyqq7GoXSOgad3fu9NRZWNukOTsf+bh5ePi8sgcHQIxHpKKVM6JFIvCpZAS5ApFEplCFVLi1eAWXCNZptdAAbkYuNAkFgY1wEymYAASrgKAAPVAwAD0RJJ0wAOhAAFQAbQARAByAC6AH4ABRC5mKADUugAlMKxeKAO4y0USxRKuUNNUSwia8UQHWEPUCqXc-kCuk4-HQV7jSbgahQYANWkIRACgB0xAdDSFjNt5MpuCpUpxNAAZjAhQBCSO+0n2x0wABkietxL98YabrwYDIUCdAD4YAkpQBvWyWji4LMgMg+-ZpuMqR0cxB84OMAC+QA)

### CODE REVIEW

File: packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx [hunk 1]
```diff
@@ -80,10 +80,11 @@ const CapacityTrendCard: React.FC = () => {
   let retentionPeriod: string = DEFAULT_PROMETHEUS_RETENTION;
 
   if (configLoaded && !configLoadError && configData?.data) {
-    const retentionRegex = /retention:\s*["'](\d+([dhmy]))["']/;
+    const retentionRegex =
+      /retention:\s*["']?((\d+y)?(\d+w)?(\d+d)?(\d+h)?(\d+m)?(\d+s)?(\d+ms)?)["']?/;
     const retentionMatch =
       configData?.data?.['config.yaml']?.match(retentionRegex);
-    if (!!retentionMatch && retentionMatch.length === 2) {
+    if (!!retentionMatch && retentionMatch.length > 0) {
```
Comment (SanjalKatiyar) [inline packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx:87]: I think `retentionMatch.length > 0` will always be true, even if match fails (check for say `retention: xyz`)...

Comment (SanjalKatiyar) [reply packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx:87]: maybe try with ??
```suggestion
    if (retentionMatch?.[1] !== "") {
```

Comment (SanjalKatiyar) [reply packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx:87]: maybe try ??
```suggestion
    if (!!retentionMatch && retentionMatch?. !== "") {
```

Comment (weirdwiz) [reply packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx]: fixed

**Comment by weirdwiz** [comment]
can you ptal @SanjalKatiyar?  forgot to ping you back after fixing the comment :p 

cc @bipuladh, can you also give it another eye? i've added the playground link to test out the matching

**Review by bipuladh — APPROVED** [review]
/lgtm

**Pull request merged**